vault/expiration.go

```diff
@@ -193,11 +193,13 @@ func (m *ExpirationManager) revokeCommon(leaseID string, force bool) error {
 	}
 
 	// Revoke the entry
-	if err := m.revokeEntry(le); err != nil {
-		if !force {
-			return err
-		} else {
-			m.logger.Printf("[WARN]: revocation from the backend failed, but in force mode so ignoring; error was: %s", err)
+	if !skipToken || le.Auth == nil {
```

Comment:
If `skipToken` is `true` and `le.Auth` is `nil`, then we'll run into the same deadlock issue.

Since we are calling this from only one place, I guess you have considered that `le.Auth` is always not `nil`.

This happens because m.lookupByToken() does not return token's lease, instead it returns only its sub-leases. I checked and it's what is happening.

It all looks fine but if `revokeCommon` is called from a different place without knowing that `le.Auth` should be `nil`, it can be a problem.